packages/insomnia/src/templating/utils.ts

```diff
@@ -291,3 +292,29 @@ export const responseTagRegex = new RegExp('{% *response *.* %}');
 export function sanitizeStrForWin32(str: string) {
   return str.replace(/\\/g, '\\\\\\\\');
 }
+
+export function getObjectUsedContextKeys<T extends Record<string, any>>(obj: T) {
```

Comment:
```suggestion
export function searchObjectForTemplateTags<T extends Record<string, any>>(obj: T) {
```
This thing could be quite slow given a large enough object, it is worht considering how general this needs to be. Do you have future plans for it or can we narrow the types a little to suit only the current use cases?

Reply:
In order to minimize the impact, I would propose the following options
Option 1: Add a new setting under _General -> Request / Response_ to enable this feature.
Option 2: Add a new setting for each request to enable this feature.

I recommend **Option 1**, as time-consuming tags defined in active environments affect all requests, not just a single one. A global setting provides clearer control and avoids impact current use cases